gadget/install/install.go

```diff
@@ -705,6 +714,15 @@ func FactoryReset(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string,
 
 	// auto-detect device if no device is forced
 	// device forcing is used for (spread) testing only
+	if bootDevice == "" {
```

Comment:
(same comments/nitpicks as above)

Reply:
Same comments from my part here as well. Maybe this should be a small helper function to reuse code.